gfp_test.go

```diff
@@ -112,6 +115,38 @@ func TestGFp(t *testing.T) {
 		}
 	})
 
+	t.Run("mul_fp_corruption", func(t *testing.T) {
+		// By enabling the mutex profiling, the go runtime will traverse the
+		// stack to measure mutex operations when a mutex is unlocked when a
+		// goroutine is blocking on it.
+		runtime.SetMutexProfileFraction(1)
+
+		var (
+			lock sync.Mutex
+			wg   sync.WaitGroup
+		)
+		wg.Add(testTimes)
+		for i := 0; i < testTimes; i++ {
+			go func() {
+				defer wg.Done()
+
+				a := togfP(randomGF(rand.Reader))
+				b := togfP(randomGF(rand.Reader))
+				c := &gfP{}
+				gfpMul(c, a, b)
+
+				lock.Lock()
+				// Make it more likely for goroutines to block on this mutex.
+				time.Sleep(time.Microsecond)
+
+				// If the frame pointer was corrupted, and another goroutine is
+				// blocked on this mutex, then this will segfault.
+				lock.Unlock()
```

Comment:
We could make this an arm specific test and include some asm code to explicitly return `R27` and `R29` so that we verify `gfpMul` doesn't modify them. I felt like that was a bit more work than I wanted to do... But this test is a bit contrived as it is right now (and it only verifies `R29` as not being corrupted)